nemo/utils/neural_type_utils.py

```diff
@@ -0,0 +1,63 @@
+from collections import defaultdict
```

Comment:
This can go into the Nemo/ core / utils directory instead of this one. Well it's just semantics it's fine here too but more appropriate there.

Reply:
done.